src/components/OrderBreakdown/LineItemFormMaybe.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-console */
+import React from 'react';
+import { Form as FinalForm, FormSpy } from 'react-final-form';
+import { injectIntl } from 'react-intl';
+import { compose } from 'redux';
+import { FieldTextInput } from '..';
+import { required, numberAtLeast, composeValidators } from '../../util/validators';
+import { formatNumber, parse } from '../FieldPhoneNumberInput/fiFormatter';
+
+import { MINIMUN_QUANTITY } from '../../config/configBookingService';
+
+import css from './LineItemFormMaybe.module.css';
+
+const FormComponent = props =>
+(<FinalForm
+  {...props}
+  render={fieldRenderProps => {
+    const { handleSubmit,
+      formName,
+      intl
+    } = fieldRenderProps;
+
+    const { quantity,  setNewQuantity } = props;
+
+    const requiredLineItem = required(intl.formatMessage({
+      id: 'OrderBreakdown.lineItemFormRequired'
+    }));
+    const requiredAtLeastOne = numberAtLeast(`${intl.formatMessage({
+      id: 'OrderBreakdown.lineItemFormAtLeast'
+    })} ${MINIMUN_QUANTITY}`, MINIMUN_QUANTITY);
+
+    const inputProps = {
+      type: 'text',
+      format: formatNumber,
+      parse: parse,
+    };
+
+    const handleChange = (formState) => {
+      if (formState.values.lineItem) {
+        setNewQuantity(formState.values.lineItem)
+      }
```

Comment:
Why do we need to check for formState.values.lineItem? shouldn't it be there when the form stat change?

Reply:
`Cannot update a component (`TransactionPageComponent`) while rendering a different component (`FormSpy`). To locate the bad setState() call inside `FormSpy``, react-dom log this warning if I do not check for `formState.values.lineItems`